scripts/Stretch.py

```diff
@@ -0,0 +1,339 @@
+# Copyright (c) 2015 Jaime van Kessel, Ultimaker B.V.
+# The PostProcessingPlugin is released under the terms of the AGPLv3 or higher.
+#
+# Christophe Baribaud 2017
+# Python implementation of https://github.com/electrocbd/post_stretch
+# Correction of hole sizes, cylinder diameters and curves
+#
+# WARNING This script has never been tested with several extruders
+#
+from ..Script import Script
+import math
+import numpy as np
+
+class GCodeStep():
+    def __init__(self,step,x,y,z,e,f,comment):
+        self.step = step
+        self.x = x
+        self.y = y
+        self.z = z
+        self.e = e
+        self.f = f
+        self.comment = comment
+
+class Stretch(Script):
+    def __init__(self):
+        super().__init__()
+
+    def getSettingDataString(self):
+        return """{
+            "name":"Post stretch script",
+            "key": "Stretch",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "stretch":
+                {
+                    "label": "Stretch distance",
+                    "description": "Distance by which the points are moved by the correction effect. The higher this value, the higher the effect",
+                    "unit": "mm",
+                    "type": "float",
+                    "default_value": 0.08,
+                    "minimum_value": 0,
+                    "minimum_value_warning": 0,
+                    "maximum_value_warning": 0.2
+                },
+                "line_width":
+                {
+                    "label": "Wall size",
+                    "description": "Average value of line width, should match roughly the corresponding parameter of Cura/CuraEngine",
+                    "unit": "mm",
+                    "type": "float",
+                    "default_value": 0.7,
+                    "minimum_value": 0.1,
+                    "maximum_value_warning": 1
+                }
+            }
+        }"""
+
+    def execute(self, data):
+        self.line_width = self.getSettingValueByKey("line_width")
```

Comment:
There is a problem, the initial value is defined after init is called. It is a "self" value because it can be different between instances of stretch objects, so what to do? Define it empty in init an overwrite it in execute?